pylint/checkers/base.py

```diff
@@ -1049,7 +1058,28 @@ def _check_using_constant_test(self, node, test):
         if not isinstance(test, except_nodes):
             inferred = utils.safe_infer(test)
 
-        if emit or isinstance(inferred, const_nodes):
+        if emit:
+            self.add_message("using-constant-test", node=node)
+        elif isinstance(inferred, const_nodes):
+            # If the constant node is a FunctionDef or Lambda then
+            #  it may be a illicit function call due to missing parentheses
+            call_inferred = None
+            if isinstance(inferred, astroid.FunctionDef):
+                call_inferred = inferred.infer_call_result()
+            elif isinstance(inferred, astroid.Lambda):
+                call_inferred = inferred.infer_call_result(node)
+            if call_inferred is not None:
```

Comment:
It might be `Uninferable` so `if call_inferred` should be enough here, as `Uninferable` has a falsey boolean value.